engine/gamesys/src/gamesys/components/comp_gui.cpp

```diff
@@ -302,6 +302,7 @@ namespace dmGameSystem
         GuiWorld* gui_world = new GuiWorld();
         if (!gui_context->m_Worlds.Full())
         {
+            gui_world->m_RenderOrder = gui_context->m_Worlds.Size();
```

Comment:
I don't think this is good enough to keep the number unique.
Scenario:
```
Spawn Coll A -> order = 0
Spawn Coll B -> order = 1
Despawn Coll A
Spawn Coll C -> order = 1
```

It would work, if the order is updated in the `CompGuiDeleteWorld()` function, and in fact we're already looping over all the worlds there.